src/lib/apputils/net-server.c

```diff
@@ -1329,7 +1336,9 @@ process_tcp_connection_read(verto_ctx *ctx, verto_ev *ev)
     } else {
         /* msglen known. */
         socklen_t local_saddrlen = sizeof(struct sockaddr_storage);
-        struct sockaddr *local_saddrp = NULL;
+        krb5_address local_addr_buf;
+        krb5_fulladdr local_addr;
```

Comment:
I don't think we can use a local variable here.  dispatch() in the KDC will call process_as_req() which will save an alias to the pointer, and possibly return (due to async kdcpreauth modules) before finish_process_as_req() uses the alias.  I think the PR needs to add fields to struct connection like it does to struct udp_dispatch_state.  (struct tcp_dispatch_state would also work, but adding fields there would be inconsistent with how we deal with remote_addr for TCP.)

(I don't like how fragile this design is with regard to object lifetimes, but the same fragility as it relates to remote_addr predates this PR.)